### DISCUSSION TITLE
Editor Needs Some Love: Let Have More Customizable Main Editor

### DISCUSSION THREAD

**Discussion by kookma** [TiddlyWiki/TiddlyWiki5#6379] [Ideas]
Ref: https://talk.tiddlywiki.org/t/query-is-there-an-existing-solution-to-preview-alignment/1903

Alignment of content / Wikitext in preview mode is an issue in my opinion!

One solution for letting manual alignment is https://talk.tiddlywiki.org/t/query-is-there-an-existing-solution-to-preview-alignment/1903/5 e.g. lets user manually align the wikitext in left (editor) with output at right side (preview pane)

This solution results in another issue, the autosize does not work anymore! It seems Tiddlywiki uses JS to resize the content area when user is typing in the edit area!

**Suggestion**:

* lets have some more customization for editor (both edit area and review pane)
* lets turn on /off alignment (e.g. scrollbars)
* lets use tiny thin scrollbars (like @pmario solution)
* lets have a button on editor toolbar to switch to fullscreen editor (JD solution)
* lets show preview pane side by side or at the bottom or top

**Comment by AnthonyMuscio** [comment]
@kookma I support these opportunities to repair and improve this. Some thoughts;

Possible Improvements
- A core feature to set floating/sticky EditorToolbar to follow you down a long tiddler.Allow easy color too.
- Somewhere to place or display information in the editor, perhaps a tag driven version of the View Templates subtitle. for example word or character counters. Or perhaps buttons to jump to bottom or top.
- A way to toggle the type and field add/edit options to appear above the text editor especially on long tiddlers. Otherwise to simply add or modify a field you have to scroll to the bottom unless the last item is enough.

Let us keep in mind the new cascade mechanism is a new way to hack this very part of tiddlywiki.

- For example by installing these two tiddlers, [top-botton-preview.json.txt](https://github.com/Jermolene/TiddlyWiki5/files/7798612/top-botton-preview.json.txt)  if you manually add a field "preview" and give it a value of top OR bottom it provides a preview (View rendered body) at the top or bottom of the editor. Perhaps we add an edit toolbar button to do this?

Here is a modified core tiddler to add items after the viewTemplates default subtitle with $:/tags/viewSubtitle
[ViewTemplateSubtitle-hack.json.txt](https://github.com/Jermolene/TiddlyWiki5/files/7798638/ViewTemplateSubtitle-hack.json.txt)

Here is a proposed core tiddler to create an editor subtitle with $:/tags/editSubtitle
[EditTemplate subtitle.json.txt](https://github.com/Jermolene/TiddlyWiki5/files/7798645/EditTemplate.subtitle.json.txt)
